src/pool.rs

```diff
@@ -421,10 +423,9 @@ impl ConnectionPool {
     /// Ban an address (i.e. replica). It no longer will serve
     /// traffic for any new transactions. Existing transactions on that replica
     /// will finish successfully or error out to the clients.
-    pub fn ban(&self, address: &Address, process_id: i32) {
-        self.stats.client_disconnecting(process_id, address.id);
-
```

Comment:
Client doesn't disconnect when server connection is banned. We send back an error message but also send `ReadyForQuery` message afterwards.

Reply:
reporting `client_disconnecting` when the client is still connected would cause all of its subsequent events to be ignored and will cause us to undercount the number of connected clients. We should only call `client_disconnecting` when we know for fact the client is disconnecting which happens only in `Drop`